src/Core/Domain/Profile/Permission/Command/UpdateModulePermissionsCommand.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Profile\Permission\Command;
+
+use PrestaShop\PrestaShop\Core\Domain\Profile\Permission\ValueObject\Permission;
+use PrestaShop\PrestaShop\Core\Domain\Profile\ValueObject\ProfileId;
+
+/**
+ * Updates module permissions for employee's profile
+ */
+class UpdateModulePermissionsCommand
+{
+    /**
+     * @var ProfileId
+     */
+    private $profileId;
+
+    /**
+     * @var int
+     */
+    private $moduleId;
+
+    /**
+     * @var Permission
+     */
+    private $permission;
+
+    /**
+     * @var bool
+     */
+    private $expectedStatus;
+
+    /**
+     * @param int $profileId
+     * @param int $moduleId
+     * @param string $permission
+     * @param bool $expectedStatus
+     */
+    public function __construct(int $profileId, int $moduleId, string $permission, bool $expectedStatus)
+    {
+        $this->profileId = new ProfileId($profileId);
+        $this->moduleId = $moduleId;
+        $this->permission = new Permission($permission);
+        $this->expectedStatus = $expectedStatus;
+    }
+
+    /**
+     * @return ProfileId
+     */
+    public function getProfileId(): ProfileId
+    {
+        return $this->profileId;
+    }
+
+    /**
+     * @return int
+     */
+    public function getModuleId(): int
+    {
+        return $this->moduleId;
+    }
+
+    /**
+     * @return Permission
+     */
+    public function getPermission(): Permission
+    {
+        return $this->permission;
+    }
+
+    /**
+     * @return bool
+     */
+    public function getExpectedStatus(): bool
```

Comment:
That's weird to use expected something in a command's accessors. When we edit the name we don't used `getExpectedName` It gives the feeling that "okay that's the status I expect or hope, but I'm not sure if it will work" 😅. (like we use in unit tests for example). It doesn't help with the readability at all and we don't even know which field matches this parameter, so I would suggest something more straight forward:

```suggestion
    public function isEnabled(): bool
```

Reply:
Or maybe:
```suggestion
    public function hasPermission(): bool
```

Reply:
The simple fact that the action is related to a boolean would suggest that maybe we should have two different commands `AddModulePermision` `RemoveModulePermission`

Reply:
I agree, I remembered having this discussion with @matks :sweat_smile: 
A "status" with a boolean is not a real status as it doesn't describe anything. I think the `hasPermission` to be in line with `getPermission` is the best idea :+1: